### ISSUE TITLE
Unable to save new custom template when using none Latin characters 

### ISSUE THREAD

**Issue by shadigaafar** [WordPress/gutenberg#41214]
### Description

When you try to add new template from post settings inside the post editor, then you try to save it, it gives you a message "saving failed". I tried to investigate the problem and it seems to me that rest url is wrong, there is double // in the middle instead of one.  **This only happens with non-latin characters.**

### Step-by-step reproduction instructions

1. Go to edit post or page
2. Go to post settings > Template
3. click "new" to add new Template, and name it with none-latin characters.
4. make changes to the template and save it. 
5. see if were able to save the template.

### Screenshots, screen recording, code snippet

![image](https://user-images.githubusercontent.com/74800166/169646942-2ef3e02f-555a-409a-be66-c13085ad478d.png)
![image](https://user-images.githubusercontent.com/74800166/169646968-995cdfd6-14b1-4668-bdf6-70d947dcf38c.png)

### Environment info

- Gutenberg  13.2.2
- Wordpresss 5.9.3

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ntsekouras** [comment]
👋 - thanks for reporting this @shadigaafar!  The problem seems to exist with our handling of non latin characters [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/components/sidebar/template/actions.js#L108). 

Can you verify that it works properly with latin characters?

**Comment by shadigaafar** [comment]
> Can you verify that it works properly with latin characters?

Yes, I tested it and it works with latin. but not with non-latin, in my case arabic.
@ntsekouras, ευχαριστώ

**Comment by ntsekouras** [comment]
@shadigaafar can you update the issue's title and description to reflect the actual bug? Thanks!

**Comment by Thelmachido** [comment]
This issue is related to https://github.com/WordPress/gutenberg/issues/41214 might be best to move the discussion there. Closing this one for now

**Comment by mburridge** [comment]
Confirming that this issue still persists in WP 6.1.1 and that it is REST related.

<img width="559" alt="image" src="https://user-images.githubusercontent.com/20643925/207585176-22d67ba6-6383-4a10-8005-cf7b2789b088.png">

To test in 6.1.1

1. Select: Appearance -> Editor in the WP Admin
2. Click the WordPress logo in top left, and select Templates
3. Click Add New button (top right)
4. Select Custom template and give it a name with at least one [non-latin character](https://en.wikipedia.org/wiki/List_of_Unicode_characters#Phonetic_scripts)
5. Save the new template
6. Confirm that you see this error:
<img width="618" alt="image" src="https://user-images.githubusercontent.com/20643925/207586749-25ee45d3-ab9a-4184-95b3-853830f21b71.png">

**Comment by Thelmachido** [comment]
@mburridge created a tracking issue to move this forward:

https://core.trac.wordpress.org/ticket/57329

Thank you 🙇‍♀️
